examples/README.md

```diff
@@ -0,0 +1,382 @@
+- Accessing detailed error information
+- Implementing retry with exponential backoff
+- Graceful degradation patterns
+- Error logging and monitoring
+- Status code handling best practices
+
+**Best for:**
+- Production applications
+- Applications requiring high reliability
+- Error monitoring integration
+- Rate limiting scenarios
+
+**Patterns covered:**
+- Retry logic for transient failures
+- Graceful degradation for non-critical errors
+- Centralized error handling
+- Integration with error tracking services
+
+---
+
+### 04. Common API Patterns
+**File:** `04_common_api_patterns.php`  
+**Difficulty:** Beginner to Intermediate
+
+Side-by-side comparison of common BambooHR API operations showing old vs new approaches.
+
+**What you'll learn:**
+- Employee directory operations
+- Updating employee information
+- Time off request management
+- Custom reports
+- File uploads and downloads
+- Batch operations
+- Webhook management
+
+**Best for:**
+- Understanding SDK v2 equivalents
+- Quick reference guide
+- Planning your migration
+- Training new developers
+
+**Covers:**
+- 7 common API patterns
+- Before/after code comparisons
+- Performance improvements
+- Best practices
+
+---
+
+### 05. Complete Application Migration
+**File:** `05_complete_application_migration.php`  
+**Difficulty:** Advanced
+
+Real-world example of migrating a complete onboarding dashboard application from old to new SDK.
+
+**What you'll learn:**
+- Structuring SDK-based applications
+- Class-based API wrappers
+- OAuth and API key support
+- Error handling at application level
+- Testing strategies
+- Performance optimization
+- Migration checklist
+
+**Best for:**
+- Large applications
+- Complex integration scenarios
+- Architecture planning
+- Team migration projects
+
+**Scenario:**
+An employee onboarding dashboard that:
+- Lists new hires
+- Shows employee information
+- Tracks time off balances
+- Manages onboarding checklists
+- Generates reports
+
+---
+
+## Running the Examples
+
+### Prerequisites
+
+1. **PHP 8.1+** is required
+   ```bash
+   php -v  # Check your version
+   ```
+
+2. **Install dependencies**
+   ```bash
+   cd /path/to/bhr-api-php
+   composer install
+   ```
+
+3. **Set environment variables**
+   
+   Create a `.env` file in the project root:
+   ```env
+   BAMBOO_COMPANY=mycompany
+   BAMBOO_API_KEY=your_api_key_here
+   
+   # For OAuth examples
+   OAUTH_CLIENT_ID=your_client_id
+   OAUTH_CLIENT_SECRET=your_client_secret
+   ```
+   
+   Or export them directly:
+   ```bash
+   export BAMBOO_COMPANY="mycompany"
+   export BAMBOO_API_KEY="your_api_key"
+   ```
+
+### Running Examples
+
+```bash
+# Run a specific example
+php examples/01_basic_api_key_migration.php
+
+# Run all examples
+for file in examples/*.php; do
+    echo "Running $file..."
+    php "$file"
+    echo ""
+done
+```
+
+## Migration Path Recommendation
+
+We recommend following this migration path:
+
+### Phase 1: Learn (Week 1)
```

Comment:
I'm a little torn about these timelines. On the one hand, they're incredibly reasonable and likely very accurate. On the other, I could see releasing the new SDK and hearing back "it's going to take us months to implement this new SDK, according to your documentation, what the heck??".

I don't know who we cater to - the developers who will sigh with relief at the time estimations they can take back to their team/manager/PM, or the leadership who might say "takes too long, we're not doing that". 

I'll just leave this stream of consciousness here...